app/controllers/storage_controller.rb

```diff
@@ -121,78 +134,43 @@ def show(record = nil)
     @lastaction = "show"
   end
 
+  def specific_buttons(pressed)
+    if BUTTON_PRESS_MAP[pressed]
+      send(BUTTON_PRESS_MAP[pressed])
+    end
 
-  # handle buttons pressed on the button bar
-  def button
-    @edit = session[:edit]                                  # Restore @edit for adv search box
-    params[:display] = @display if %w(all_vms vms hosts).include?(@display) # Were we displaying vms or hosts
+    handle_host_power_buttons(pressed)
 
-    if params[:pressed].starts_with?("vm_", # Handle buttons from sub-items screen
-                                     "miq_template_",
-                                     "guest_",
-                                     "host_")
+    BUTTON_PRESS_MAP.keys.include?(pressed)
+  end
 
-      scanhosts if params[:pressed] == "host_scan"
-      analyze_check_compliance_hosts if params[:pressed] == "host_analyze_check_compliance"
-      check_compliance_hosts if params[:pressed] == "host_check_compliance"
-      refreshhosts   if params[:pressed] == "host_refresh"
-      tag(Host) if params[:pressed] == "host_tag"
-      assign_policies(Host) if params[:pressed] == "host_protect"
-      edit_record  if params[:pressed] == "host_edit"
-      deletehosts if params[:pressed] == "host_delete"
+  def handle_host_power_buttons(pressed)
+    power_events = %w(
```

Comment:
I just made the same list here:

https://github.com/ManageIQ/manageiq-ui-classic/pull/1456/files#diff-e9b783f7735ad4920f9efe6512c42befR20